core/imageroot/usr/local/sbin/apply-vpn-routes

```diff
@@ -0,0 +1,147 @@
+skip_onetworks = []
+ipaddr_proc = subprocess.run(['ip', '-j', 'address', 'show'], stdout=subprocess.PIPE)
+ipaddr_reply = json.loads(ipaddr_proc.stdout)
+for iface in ipaddr_reply:
+    if not 'addr_info' in iface:
+        continue
+    for iaddr in iface['addr_info']:
+        skip_onetworks.append(ipm.ip_network(f"{iaddr['local']}/{iaddr['prefixlen']}", strict=False))
+
+errors = 0
+valid_destinations = []
+for node_id in destmap:
+    # Apply immediately the new configuration to the WireGuard wg0 interface...
+    wset_proc = agent.run_helper('wg', 'set', 'wg0', 'peer', peers[node_id]["public_key"],
+        'persistent-keepalive', '25',
+        'allowed-ips', ','.join(destmap[node_id]),
+        *get_wgset_endpoint_clause(peers[node_id].get('endpoint', '')),
+        log_command=True,
+    )
+    if wset_proc.returncode != 0:
+        errors +=1
+        print(agent.SD_ERR + f'Runtime change of allowed-ips has failed for peer node/{node_id}', file=sys.stderr)
+
+    # ...and to the system routing table:
+    for xdest in destmap[node_id]:
+        push_route = False
+
+        odest = ipm.ip_address(xdest) # object form of xdest for network calculations
+        if odest in cluster_onetwork:
+            push_route = True # xdest is in the cluster network, push it
+        else:
+            for skip_onetwork in skip_onetworks:
+                if odest in skip_onetwork:
+                    push_route = False
+                    break # xdest belongs to a network with direct interface link, skip it
+            else:
+                push_route = True # xdest belongs to a remote network, push it
+
+        if not push_route:
+            print(agent.SD_INFO + f'Address {xdest} is not routed through VPN', file=sys.stderr)
+            continue
+
+        # Yes, let's route xdest through VPN:
+        ipreplace_proc = agent.run_helper('ip', 'route', 'replace', xdest, 'nexthop', 'dev', 'wg0', log_command=True)
+        if ipreplace_proc.returncode != 0:
+            errors += 1
+            print(agent.SD_ERR + f"Runtime change of routing table has failed for destination {xdest}", file=sys.stderr)
+
+        valid_destinations.append(xdest.removesuffix('/32'))
+
+#
+# Purge the route table from stale entries
+#
+iproute_proc = subprocess.run(['ip', '-j', 'route', 'list', 'dev', 'wg0'], stdout=subprocess.PIPE)
+if iproute_proc.returncode == 0:
+    routes = json.loads(iproute_proc.stdout)
+    for route_object in routes:
+        ydest = route_object['dst']
+        if not ydest in valid_destinations:
+            ipdelete_proc = agent.run_helper('ip', 'route', 'delete', ydest, log_command=True)
+            if ipdelete_proc.returncode != 0:
+                print(agent.SD_ERR + f"Error while removing the stale destination {ydest}", file=sys.stderr)
+                errors += 1
+else:
+    print(agent.SD_ERR + f"Error while reading wg0 routes", file=sys.stderr)
+    errors += 1
```

Comment:
Are you sure the whole thing is needed?
wg-quick can already handle simple cases like this one: https://man7.org/linux/man-pages/man8/wg-quick.8.html

Reply:
The wg-quick script handles everything for you, also extra routes for local networks.

Reply:
Recap of the private discussion:
- Samba and OpenLDAP binds to the wg IP, if the VPN is restarted using `wgquick`, the service will fail
- actual configuration is hybrid and partially relays on wgquick script
- the VPN now routes only Samba IP addresses, no extra routes are allowed

Let's keep current implementation, but we should remember to document how the things work.